Source/WebCore/workers/WorkerGlobalScope.cpp

```diff
@@ -466,6 +470,9 @@ void WorkerGlobalScope::addConsoleMessage(std::unique_ptr<Inspector::ConsoleMess
     if (UNLIKELY(settingsValues().logsPageMessagesToSystemConsoleEnabled && sessionID && !sessionID->isEphemeral()))
         PageConsoleClient::logMessageToSystemConsole(*message);
 
+#if ENABLE(WEBDRIVER_BIDI)
+    AutomationInstrumentation::addMessageToConsole(message);
```

Comment:
this might not happen on the main thread.  is that a problem?  or is `WebAutomationSessionProxy` threadsafe?

Reply:
IIUC, `WebAutomationSessionProxy` is not thread-safe by itself (e.g. it inherits from `RefCounted` instead of `ThreadSafeRefCounted`, etc.).

But its `addMessageToConsole` method just forwards its parameters to the respective `WebAutomationSession` on the UIProcess through `WebProcess::singleton()->parentProcessConnection()->send(...)`, without accessing other instance data. In the header, `send()` is documented as thread-safe. Is this enough?

Maybe in the future, when instrumenting something else, `WebAutomationSessionProxy` will actually need to be made thread-safe?

Reply:
but is `WebProcess::singleton` threadsafe?

perhaps just in case you could have a `ensureOnMainThread` somewhere?

Reply:
That's an option.

Given `Inspector::ConsoleMessage` is noncopyable and `WebAutomationSession::addMessageToConsole` (the client function) already unpacks the information from it before passing ahead to the `UIProcess`, I'll update the patch to make `AutomationInstrumentation::addMessageToConsole` extract the information, on the thread where we get it, passing to the main thread (and further down the pipeline) a simplified object, with just the streamlined information we need. Does this sound ok?